libyul/AsmParser.h

```diff
@@ -96,7 +126,10 @@ class Parser: public langutil::ParserBase
 
 private:
 	Dialect const& m_dialect;
-	std::optional<langutil::SourceLocation> m_locationOverride;
+
+	std::optional<std::map<unsigned, std::shared_ptr<langutil::CharStream>>> m_charStreamMap;
+	langutil::SourceLocation m_locationOverride;
```

Comment:
Can you change this to a `shared_ptr`? If it stays like this, we will create tons of different shared pointers pointing at equivalent data.